### ISSUE TITLE
Several of my images are hashing to the same value

### ISSUE THREAD

**Issue by dazerdude** [empira/PDFsharp#197]
If you think there is a bug in PDFsharp then please use the IssueSubmissionTemplate to make the issue replicable.  
https://docs.pdfsharp.net/General/Issue-Reporting.html

Thanks.

# Resources

The official project web site:  
https://docs.pdfsharp.net/

The official peer-to-peer support forum:  
http://forum.pdfsharp.net/

# Reporting an Issue Here
## Expected Behavior
These are different images, but they are being deduped. 

After load, before writing: 
```
engine.ImageLib.DiceNumbers[0].XImage._path
"images/cache//images/color/dice/numbers/1.png"
engine.ImageLib.DiceNumbers[1].XImage._path
"images/cache//images/color/dice/numbers/2.png"
```

After writing:
```
engine.ImageLib.DiceNumbers[0].XImage._path
"*md5:F5F24B666FEA7CC6825CEAE8F9750362"
engine.ImageLib.DiceNumbers[1].XImage._path
"*md5:F5F24B666FEA7CC6825CEAE8F9750362"
```

Files in question:
![1](https://github.com/user-attachments/assets/bc115f5d-3940-4102-8e47-df7eab931217)
![2](https://github.com/user-attachments/assets/f233ef28-a630-4cbf-afc1-3349abe5b396)

## Actual Behavior
They should not be deduped. Consider using a stronger hash algorithm than MD5 that isn't so prone to collide. Possibly you're not hashing the full file data, and instead some subset that happens to be identical.

Alternatively, you could provide us a way to override this behavior with a selector factory. Currently as it's computed on every XImage instance directly, and overrides XImage._path, this is not possible w/o some real reflection work. I've ended up with this, which is a very unfortunate solution.

```
private static XImage LoadImageAndForceNoDupes(string path)
{
    var image = XImage.FromFile(path);
    var privateFlags = BindingFlags.NonPublic | BindingFlags.Instance;
    var selectorField = image.GetType().GetField("_selector", privateFlags)!;
    var selector = selectorField.FieldType.GetConstructor([typeof(XImage), typeof(PdfDocumentOptions)])!.Invoke([image, null]);
    selector.GetType().GetField("<Path>k__BackingField", privateFlags)!.SetValue(selector, Guid.NewGuid().ToString());
    selectorField.SetValue(image, selector);
    return image;
}
```

In my code, every XImage instance is built from a unique file and reused, so this works well for me. Some way to configure this behavior in the library w/o using all this reflection would be greatly appreciated. I just don't need any deduping at all beyond the instance pointer, but it doesn't appear like there is any way to disable the hashing functions in ImageSelector.

## Steps to Reproduce the Behavior
We strongly recommend using the IssueSubmissionTemplate to make sure we can replicate the issue.  
https://docs.pdfsharp.net/General/Issue-Reporting.html

**Comment by ThomasHoevel** [comment]
Looks like a duplicate of issue 127. Fix will be in 6.2.0 Preview 2.

**Comment by dazerdude** [comment]
Glad to hear it's a KI. When will Preview 2 be available?

Is that code public yet? ImageSelector looks largely similar to what I have locally.
